lib/datadog/appsec/security_engine/runner.rb

```diff
@@ -9,9 +9,10 @@ module SecurityEngine
       class Runner
         SUCCESSFUL_EXECUTION_CODES = [:ok, :match].freeze
 
-        def initialize(waf_context)
+        def initialize(waf_handle)
           @mutex = Mutex.new
-          @waf_context = waf_context
+          @waf_handle = waf_handle
+          @waf_context = @waf_handle.build_context
```

Comment:
If we know that util runner is done waf handle will exist, we can defer that call to a method with memoization.

Reply:
done in [761c125](https://github.com/DataDog/dd-trace-rb/pull/4833/commits/761c125cf9cb514600144decd744a24e69e265c0)